sdk/eventgrid/azure-eventgrid/swagger/postprocess_eventnames.py

```diff
@@ -1,44 +1,41 @@
-    'IoTHubDeviceDisconnectedEventName': "Microsoft.Devices.DeviceDisconnected",
-    'ResourceDeleteFailureEventName': "Microsoft.Resources.ResourceDeleteFailure",
-    'ResourceDeleteCancelEventName': "Microsoft.Resources.ResourceDeleteCancel",
-    'AcsChatThreadParticipantAddedEventName': "Microsoft.Communication.ChatThreadParticipantAdded",
-    'ResourceDeleteSuccessEventName': "Microsoft.Resources.ResourceDeleteSuccess",
-    'EventGridSubscriptionValidationEventName': "Microsoft.EventGrid.SubscriptionValidationEvent",
-    'ResourceWriteSuccessEventName': "Microsoft.Resources.ResourceWriteSuccess",
-    'ResourceActionSuccessEventName': "Microsoft.Resources.ResourceActionSuccess",
-    'ResourceWriteCancelEventName': "Microsoft.Resources.ResourceWriteCancel",
-    'ResourceActionFailureEventName': "Microsoft.Resources.ResourceActionFailure",
-    'AcsChatMemberRemovedFromThreadWithUserEventName': "Microsoft.Communication.ChatMemberRemovedFromThreadWithUser",
-    'IoTHubDeviceConnectedEventName': "Microsoft.Devices.DeviceConnected",
-    'EventGridSubscriptionDeletedEventName': "Microsoft.EventGrid.SubscriptionDeletedEvent",
-    'AcsChatThreadParticipantRemovedEventName': "Microsoft.Communication.ChatThreadParticipantRemoved",
-    'ResourceActionCancelEventName': "Microsoft.Resources.ResourceActionCancel",
-    'IoTHubDeviceCreatedEventName': "Microsoft.Devices.DeviceCreated"
-}
+#===============================================deprecated===========================================
+# def event_tuples(system_events):
+#     tup_list = []
+#     for event in system_events:
+#         class_name = "Name".join(event[0].rsplit('Data', 1))
+#         try:
+#             event_name = re.findall("Microsoft.[a-zA-Z]+.[a-zA-Z]+", event[1].__doc__)[0]
+#         except:
+#             # these two are just superclasses and are known exceptions.
+#             if event[0] not in ('ContainerRegistryArtifactEventData', 'ContainerRegistryEventData'):
+#                 warnings.warn("Unable to generate the event mapping for {}".format(event[0]))
+#                 sys.exit(1)
+#         tup_list.append((class_name, event_name))
+#     return tup_list
+#===============================================deprecated===========================================
 
-def event_tuples(system_events):
-    tup_list = []
-    for event in system_events:
-        class_name = "Name".join(event[0].rsplit('Data', 1))
-        try:
-            event_name = re.findall("Microsoft.[a-zA-Z]+.[a-zA-Z]+", event[1].__doc__)[0]
-        except:
-            # these two are just superclasses and are known exceptions.
-            if event[0] not in ('ContainerRegistryArtifactEventData', 'ContainerRegistryEventData'):
+def extract(definitions):
+    if not definitions:
+        return
+    tups = []
+    for event in definitions:
+        if event.endswith('Data') and event not in ('ContainerRegistryArtifactEventData', 'ContainerRegistryEventData'):
```

Comment:
I assume `('ContainerRegistryArtifactEventData', 'ContainerRegistryEventData')` is some kind of exception list, I would make it a more explicit configuration outside of the code for clarity